[cs] Add using statements via @:csUsing

kLabz opened the pull request:
~~TODO: make sure it's added to the beginning of the file~~

kLabz committed:
[cs] Add using statements via @:csUsing

back2dos commented:
Don't we have `@:cs.using` these days?

Simn commented:
Please update to use `@:cs.using` instead. With that I'm fine merging this if you still think it's useful.

kLabz committed:
[cs] Only allow @:cs.using on first type of the module

Otherwise we get a cs compilation error about using clauses needing to
preced all other elements defined in the namespace.

kLabz committed:
[cs] Use @:cs.using instead of @csUsing

kLabz commented:
I'll add some tests later today

kLabz committed:
Add test for @:cs.using

kLabz committed:
Fix gencs.ml
